File: src/com/colacelli/irclib/connection/listeners/OnCtcpListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.messages.CTCPMessage;

public interface OnCtcpListener extends Listener {
    Type TYPE = Type.CTCP;

    String CTCP_CHARACTER = "\u0001";

    void onCtcp(Connection connection, CTCPMessage message, String... args);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/com/colacelli/irclib/connection/listeners/OnCTCPListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.messages.CTCPMessage

interface OnCTCPListener : Listener {
    fun onCTCP(connection: Connection, message: CTCPMessage, vararg args: List<String>)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.CONNECT
        const val CTCP_CHARACTER = "\u0001"
    }
}
